File: 卡码/代码随想录/1/08-贪心算法/C++/03-困难/LeetCode_0452.cpp
//452. 用最少数量的箭引爆气球
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
	int findMinArrowShots(vector<vector<int>>& points) {
		if(points.empty()) return 0;
		sort(points.begin(),points.end(),[](vector<int>& p1,vector<int>& p2){
			return p1[0]<p2[0];
		});
		int arrow=1;
		for(int i=1;i<points.size();i++){
			if(points[i][0]>points[i-1][1]) arrow++;
			else points[i][1]=min(points[i-1][1],points[i][1]);
		}
		return arrow;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_03_困难/LeetCode_0452.java
//452. 用最少数量的箭引爆气球
package com.kurumi.greedy._03_困难;

import java.util.Arrays;

public class LeetCode_0452 {

}
class Solution {
  public int findMinArrowShots(int[][] points) {
    if(points.length==0) return 0;
    int arrow=1;
    Arrays.sort(points,(p1,p2)->Integer.compare(p1[0], p2[0]));
    for(int i=1;i<points.length;i++){
      if(points[i][0]>points[i-1][1]) arrow++;
      else{
        points[i][1]=Integer.min(points[i][1], points[i-1][1]);
      }
    }
    return arrow;
  }
}
